### DISCUSSION TITLE
Add entity linker from another source to model

### DISCUSSION THREAD

**Discussion by farrandi** [explosion/spaCy#8608] [Help: Coding & Implementations] [answered]
Hey! I am trying to add the `UmlsEntityLinker from scispacy.umls_linking` to my trained model. However I am not clear on how to do this.
I read the documentation [here](https://spacy.io/api/entitylinker) and I tried doing this:
```Python:
from scispacy.umls_linking import UmlsEntityLinker
nlp = spacy.load("models/blank6/full-model")
entity_linker = UmlsEntityLinker(nlp.vocab, nlp)
```

However, I am getting this error and the process seems to keep running (I am using Jupyter notebook):
```
/home/farrandi/.local/lib/python3.8/site-packages/sklearn/base.py:310: UserWarning: Trying to unpickle estimator TfidfTransformer from version 0.20.3 when using version 0.24.2. This might lead to breaking code or invalid results. Use at your own risk.
  warnings.warn(
/home/farrandi/.local/lib/python3.8/site-packages/sklearn/base.py:310: UserWarning: Trying to unpickle estimator TfidfVectorizer from version 0.20.3 when using version 0.24.2. This might lead to breaking code or invalid results. Use at your own risk.
  warnings.warn(
```

Is there a way to add this entity_linker to the pipeline?

**Comment by svlandeg** [comment, marked as answer]
Hi! 

The `UmlsEntityLinker` is not a built-in functionality from spaCy, but provided by the [ScispaCy](https://github.com/allenai/scispacy) project, so you'll have to follow the documentation from there. It is not compatible with spaCy's `entity_linker`. That said, if it's a "normal" pipeline component, you should be able to fetch a pretrained one with `get_pipe` or create a new one with `add_pipe`, but there might be other requirements I'm not aware of. If you run into issues, it's probably better to consult the authors of this package over at https://github.com/allenai/scispacy/issues.

**Reply by farrandi** [reply]
Oh alright got it!
